proto/pbft.proto

```diff
@@ -89,4 +117,39 @@ message pbft_peer_info
     uint32 http_port = 3;
     string name = 4;
     string uuid = 5;
+    pbft_request_type type = 6;
+    oneof msg
+    {
+        database_msg operation = 7;
+        pbft_internal_msg command = 8;
+    }
+    uint64 timestamp = 9;
+    string client = 10;
+}
+
+enum pbft_internal_msg_type {
+    PBFT_IMSG_UNDEFINED = 0;
+    PBFT_IMSG_NEW_CONFIG = 1;
+}
+
+enum pbft_request_type {
+    PBFT_REQ_UNDEFINED = 0;
+    PBFT_REQ_DATABASE = 1;
+    PBFT_REQ_NEW_CONFIG = 2;
+    PBFT_REQ_INTERNAL = 3;
+    PBFT_REQ_NO_OP = 4;
+}
+
+message pbft_internal_msg
+{
+    pbft_internal_msg_type type = 1;
+
+    // for new_config
+    string configuration = 2;
+}
+
+message prepared_proof
+{
+    bytes pre_prepare = 1;  // O
+    repeated bytes prepare = 2; // P, P_m
```

Comment:
bzn_envelope

Reply:
Changed to:

message prepared_proof
{
    bzn_envelope pre_prepare = 1;  // O
    repeated bzn_envelope prepare = 2; // P, P_m
}